src/Presentation/Presenters/MembershipApplicationConfirmationHtmlPresenter.php

```diff
@@ -63,7 +65,16 @@ private function getConfirmationPageArguments( Application $membershipApplicatio
 			'bankData' => $this->getBankDataArguments( $membershipApplication->getPayment()->getPaymentMethod() ),
 			'payPalData' => $this->getPayPalDataArguments(
 				$membershipApplication->getPayment()->getPaymentMethod()
-			)
+			),
+			'urls' => [
```

Comment:
But in this case i specifically need to generate a URL with the donation ID and the update token and I cannot do it that way, can I?

```
'cancelMembership'  => $this->urlGenerator->generateRelativeUrl(
	'CancelMembership',
	[
		'donationId' => $membershipApplication->getId(),
		'updateToken' => $updateToken,
	]
)
```

Either way, the main reason I did not do this was because it would require some annoying refactoring of the test where all URLs would need to be added to the respective PHPUnit test for no added benefit (YAGNI).